connectivity/http/http.c

```diff
@@ -327,16 +331,41 @@ static int request_log(void *cls, const struct sockaddr *addr, socklen_t addrlen
   return MHD_YES;
 }
 
+static status_t build_request(ta_http_request_t *req, const char *data, size_t size) {
+  if (size > 0) {
+    if (size + req->request_len >= MAX_REQUEST_LEN) {
+      req->answer_string = strdup(STR_HTTP_REQUEST_SIZE_EXCEED);
+      req->answer_code = MHD_HTTP_INTERNAL_SERVER_ERROR;
+      return SC_HTTP_INTERNAL_SERVICE_ERROR;
+    }
+    char *request = malloc(req->request_len + size + 1);
+    if (request == NULL) {
+      req->answer_string = strdup(STR_HTTP_INTERNAL_SERVICE_ERROR);
```

Comment:
If OOM happened in line 341, then line 343 may meet the same problem. I think solve this problem here may be hard. If so, could you at least at TODO here to remind us in the future.

Reply:
Get it.

Reply:
How about modifying the `STR_HTTP_INTERNAL_SERVICE_ERROR` to a `const char*`?

Reply:
In the example program of [libmicrohttpd/tutorial](https://www.gnu.org/software/libmicrohttpd/tutorial.html#Example-programs)
The response pages are declared as `const char*` instead of macros.

We need to modify all the HTTP answer strings if apply this modification.

Reply:
So we are going to reuse the `const char*` variable as a read-only variable?

Reply:
Most of our HTTP APIs encounter this problem.
Let's discuss and do it in another issue and pull request.

Reply:
OK